securedrop_client/queue.py

```diff
@@ -139,27 +153,38 @@ def __init__(self, api_client: API, session_maker: scoped_session) -> None:
 
         self.main_thread = QThread()
         self.download_file_thread = QThread()
+        self.metadata_thread = QThread()
 
         self.main_queue = RunnableQueue(api_client, session_maker)
         self.download_file_queue = RunnableQueue(api_client, session_maker)
+        self.metadata_queue = RunnableQueue(api_client, session_maker, size=1)
 
         self.main_queue.moveToThread(self.main_thread)
         self.download_file_queue.moveToThread(self.download_file_thread)
+        self.metadata_queue.moveToThread(self.metadata_thread)
 
         self.main_thread.started.connect(self.main_queue.process)
         self.download_file_thread.started.connect(self.download_file_queue.process)
+        self.metadata_thread.started.connect(self.metadata_queue.process)
 
         self.main_queue.paused.connect(self.on_queue_paused)
         self.download_file_queue.paused.connect(self.on_queue_paused)
+        self.metadata_queue.paused.connect(self.on_queue_paused)
+
+        self.main_queue.pinged.connect(self.resume_queues)
```

Comment:
you can remove the main_queue and download_file_queue pinged signal also because there is already a resume signal emitted by the queue manager, see https://github.com/freedomofpress/securedrop-client/blob/33856fb8e12344c323c0185b4c0536f90788b33c/securedrop_client/queue.py#L177-L181

so once you remove these three lines of code, and add self.resume_queues()` in [on_sync_success](https://github.com/freedomofpress/securedrop-client/blob/master/securedrop_client/logic.py#L412), and update the queue manager's resume function to check if a queue is paused before emitting the resume signal, then this PR should be good to go.

Reply:
again, this is what closes #672, see https://github.com/freedomofpress/securedrop-client/pull/715#issuecomment-577260279

Reply:
OK... I've made the changes you've requested, updated the unit tests, and rebased with latest master. :+1: